exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -344,3 +348,25 @@ func (prwe *PrwExporter) execute(ctx context.Context, writeReq *prompb.WriteRequ
 	}
 	return nil
 }
+
+// FormatHeaderValues will add the name/version pair to the User-Agent request
+// header. If the extra parameters are provided they will be added as metadata to the
+// name/version pair resulting in the following format.
+// "name/version (extra0; extra1; ...)"
+// The User-Agent part will be concatenated with this current request's User-Agent string.
+func FormatHeaderValues(req *http.Request, name, version string, extra ...string) {
```

Comment:
The name doesn't seem to match what this function actually does.  I might have it simply perform the formatting and return a string, leaving it to `AddToUserAgent` to interact with the `*http.Request`.  It also appears that these functions don't need to be exported, so I wouldn't.  I try always to start with unexported identifiers and only export them when I find a compelling need to do so.

```suggestion
func formatHeaderValues(name, version string, extra ...string) string {
```

Then it would be used like this:

```go
addToUserAgent(req, formatHeaderValues("name", "version", "extra", "info", "here"))
```

Reply:
Yes I will change the function return type and change the implementation accordingly